### PR TITLE
Add optional "duck-typing" of custom data to allow numeric/boolean comparisons

### PR DESCRIPTION
**Pull request by frankus**
The Apptentive custom data system allows strings, numbers, and boolean values to be compared using operators appropriate for each type. 

This PR adds an option to have the Apptentive mParticle kit automatically copy "number-like" and "boolean-like" custom data strings received from mParticle to a second custom data key (with a suffix) to allow boolean and numeric comparisons wherever possible. 

It also adds support for passing screen events to the Apptentive SDK.

### CODE REVIEW

**Comment by alexs-mparticle** [comment]
Tracked Internally as 67652

File: src/main/java/com/mparticle/kits/ApptentiveKit.java [hunk 1]
```diff
@@ -170,20 +184,18 @@ public void removeUserIdentity(MParticle.IdentityType identityType) {
 
 	@Override
 	public List<ReportingMessage> logEvent(MPEvent event) {
-		Map<String, String> customData = event.getInfo();
-		if (customData != null) {
-			Apptentive.engage(getContext(), event.getEventName(), Collections.<String, Object>unmodifiableMap(customData));
-		} else {
-			Apptentive.engage(getContext(), event.getEventName());
-		}
+		engage(getContext(), event.getEventName(), event.getInfo());
```
Comment (willpassidomo) [inline src/main/java/com/mparticle/kits/ApptentiveKit.java:187]: nitpick: `MPEvent#getInfo()` is deprercated, use `getCustomAttributes()` instead

**Review by willpassidomo — APPROVED** [review]
Looks good to me! Just had one little nit about a deprecated API, but I won't let that require a re-review. 

One thing I just want to make note of and ensure you're aware of; this code defaults `enableTypeDetection` to `true` so this is going to slightly change behavior for everyone currently using the kit once they upgrade. I'm sure you know better than I if this will have any downstream effects in Apptentive, just want to call it out

File: src/main/java/com/mparticle/kits/ApptentiveKit.java (hunk 1 above)
Comment (melj32app) [reply src/main/java/com/mparticle/kits/ApptentiveKit.java:187]: Yes, @willpassidomo, we intentionally set the flag to true as we believe that will be the best experience for customers.  In future, we may ask you to allow customers to turn it off on their connection.  We'll track the deprecated api usage for a future release.

Comment (willpassidomo) [reply src/main/java/com/mparticle/kits/ApptentiveKit.java:187]: sounds good! Good to merge when you're ready :)

**Pull request merged**